### DISCUSSION TITLE
Questions about statistics

### DISCUSSION THREAD

**Discussion by astaaudzi** [fishsizeproject/CPUEcourse#2] [Q&A]
In this group you can ask statistical questions related to the course content. We will aim to answer them to the best of our capacity

**Comment by martinogonowski** [comment]
GLM with Gamma distribution would probably be more appropriate for the trout model but I seldom get such models to run. Any ideas on why this might be? I often get weird error messages and convergence issues. Is this normal? Any ideas?

**Reply by carlsmith66** [reply]
Possibly a few zeros in the response variable?

**Reply by martinogonowski** [reply]
Nope but if I find that specific example I can share it.

**Comment by martinogonowski** [comment]
Perhaps a question for later lectures but is it possible to separate random interannual variability from true population variability over time? 

For example, would it make sense to run a model like mod <- lm(Catch ~YearF + Year) where YearF is categorical and Year is continuous? I'm aware there is interannual dependency here but let's assume we have a small data set with <10 years like the bitterling data.

**Reply by carlsmith66** [reply]
Perhaps - though I have not tried this. Perhaps also include fYear as random term and year as fixed?

**Reply by martinogonowski** [reply]
Ok. Good suggestion. I'll try that. Thank you Carl!

**Reply by astaaudzi** [reply]
Yes, we ran some models like that too (in a Bayesian framework, coded with stan), where year was considered as a fixed term and also as a random term (https://www.nature.com/articles/s41559-020-1171-0), see code here https://github.com/astaaudzi/RLSfishSize

**Comment by carlsmith66** [comment]
We got similar results for Polish sticklebacks with a Bayesian temporal-spatial model

Sent with [Proton Mail](https://proton.me/) secure email.

------- Original Message -------

**Comment by HEThatcher** [comment]
### Standardising catch using multiple categorical variables 

Hi both, firstly thanks very much for running this course, as my first introduction to modelling cpue I found it particularly valuable. 

I'm trying to standardise catches of lobsters at specific locations within a windfarm as a result of effort (Number of pots = 1-24), year (2016-2022), and sediment type (either hard or soft). Although I have been able to use the simple linear model code to predict catches of lobster as a result of these factors. I'm struggling to decide if the data is suited (is there a linear response) for this type of analysis, and if it is, is there a better model I can use. A weird pattern in the residuals vs fitted is throwing me off. 

Below is a plot of the effect of effort on catches (from raw data) and the residuals vs fitted for the model with the lowest AIC score. 

Any guidance would be much appreciated. If further detail on the dataset is required I can provide. 
Many thanks.
![Screenshot 2022-11-29 181633](https://user-images.githubusercontent.com/106250250/204613160-21ddb9b8-963f-4b32-80cc-d8e8c3eaee33.png)
![Screenshot 2022-11-29 181826](https://user-images.githubusercontent.com/106250250/204613427-f737bbe0-b89e-4246-81fb-9a04918445c0.png)

**Comment by carlsmith66** [comment]
It looks like you have quite a few zeros in the response variable - did you model these data with a Gaussian distribution? There also seems to be some non-linearity in the residuals plot (especially for high fitted values). You could confirm the non-linearity with a GAM.

A solution might be a zero-adjusted model, or perhaps a Tweedie distribution.

**Reply by HEThatcher** [reply]
Thanks for getting back to me. 

Zeros made up 9% of the total values, although there were a high proportion of 1s & 2s (histogram below). For this reason I transformed the data using sqrt in the lm call (code below) and did not change the distribution. 

As you suggested I've used the GAM code to inspect non-linearlity, and it looks like there are non linear relationships present in the data. Would this suggest I should try using a non-linear regression or is there another type of distbution that would be suitable? Or should I use a GAM to model the data? 

Again, many thanks for helping with this. 

model call = mod1 <- lm(sqrt(Total) ~ Effort + Year + Scour,
           data=lobster1)  

![Screenshot 2022-12-01 115517](https://user-images.githubusercontent.com/106250250/205046778-e4e7d12f-5a3b-482b-af5c-8b9b6907efc8.png)

**Comment by carlsmith66** [comment]
I guess total lobster is a count, in which case an lm is not appropriate here; these are count data. Don't transform your response variable and instead fit a Poisson GLM. You will need to check for overdispersion. If the model is overdispersed, there are a few model checks to run through to identify the source.  Depending on what these show, you can deal with the overdispersion appropriately.

If you are struggling with any of this, feel free to email (carl.smith@biol.uni.lodz.pl) me your data and script and I will take a look at it.
